### PR TITLE
Added possibility to enable ultimate exchange outside of trials

### PR DESCRIPTION
**Pull request by memus**
I don't know if you approve this one but i know many players want this feature. If you want me to change something i'll close pull request and fix that.

### CODE REVIEW

**Comment by kyoma** [comment]
It's a good idea and I've been thinking of doing it as well. Will look more closely at the code tonight, I might change how the ultimate window is managed to avoid having to register the raid-specific things just for the ultimate window. 

I might actually end up releasing it as a seperate addon but that's more of an idea than an actual plan.

File: RaidNotifier.lua [hunk 1]
```diff
@@ -318,6 +319,26 @@ do ----------------------
...
 	end
 
+	function RaidNotifier:ShowUltimatesEverywhere(force)
+		if (self.raidId == nil) then
+			self.raidId = self:GetRaidIdFromCurrentZone()
+		end
+
+		if (force) then
+			if (self.raidId > 0) then
+				self:RegisterForUltimateChanges()
```
Comment (memus) [inline RaidNotifier.lua]: Already inside trial so just register ultimates only.

File: RaidNotifier.lua [hunk 2]
```diff
@@ -318,6 +319,26 @@ do ----------------------
...
+			else
+				self:RegisterEvents(true)
```
Comment (memus) [inline RaidNotifier.lua]: Register only things required for ultimates.

File: RaidNotifier.lua [hunk 3]
```diff
@@ -337,12 +358,12 @@ do ----------------------
 			p("Enable Ultimate Exchange")
 			settings.enabled = true
 			settings.hidden = false
-			self:RegisterForUltimateChanges()
+			self:ShowUltimatesEverywhere(true)
```
Comment (memus) [inline RaidNotifier.lua]: if outside trial - temporary enable ultimates (disabled after relog, reloadui etc)

File: RaidNotifier.lua [hunk 4]
```diff
@@ -423,60 +444,63 @@ do ----------------------
 	end
 
 	local listening = false
-	function RaidNotifier:RegisterEvents(raidId)
+	function RaidNotifier:RegisterEvents(forceUltimate, raidId)
```
Comment (memus) [inline RaidNotifier.lua]: force ultimate and raidId == 0 will register only things required for ultimates

**Comment by memus** [comment]
I was thinkig about separated addon but then i realized that this addon need to checkout if raidnotifier ultimates are working and disable it or something. 
There are 2 possibilities to enable ultimates outside:
* temporary - by typing /rnulti enable
* always - RN settings.

> I might change how the ultimate window is managed to avoid having to register the raid-specific things
Thats why i did some changes to avoid that but take a look and give me a review later :)

**Comment by kyoma** [comment]
Yes I saw the code, my point is just if I move the Ultimate Window in the xml outside of our RaidNotifierUI element we don't need to modify any registration code and just show/hide the window outside of raids. :)

**Comment by memus** [comment]
Some kind of wrapper without messing raid notifier code?

**Comment by kyoma** [comment]
Exactly. Then all that is needed is calling RegisterForUltimateExchange since that code is already seperated from the rest of the raid code.

**Pull request closed without merging**